tests/suite/cli_rustup.rs

```diff
@@ -2954,3 +2954,68 @@ async fn warn_on_duplicate_rust_toolchain_file() {
         )
         .await;
 }
+
+#[tokio::test]
+async fn custom_toolchain_with_components_toolchains_profile_does_not_err() {
+    let cx = CliTestContext::new(Scenario::SimpleV2).await;
+
+    let cwd = cx.config.current_dir();
+    let toolchain_file = cwd.join("rust-toolchain.toml");
+
+    // install a toolchain so we can make a custom toolchain that links to it
+    cx.config
+        .expect_stderr_ok(
+            &["rustup", "toolchain", "install", "nightly"],
+            &format!(
```

Comment:
We have an extra `for_host!()` macro here. It's far from idiomatic but using consistent APIs will help us with further evolution of our testing code. Thanks!